micro_sam/sam_annotator/_widgets.py

```diff
@@ -655,7 +681,76 @@ def _create_settings_widget(self):
         settings = _make_collapsible(setting_values, title="Settings")
         return settings
 
+    def _validate_inputs(self):
+        """Validates inputs and returns a dictionary for message generation."""
+
+        if self.embeddings_save_path is not None:
+            # Validate image data signature
+            image = self.image_selection.get_value()
+            img_signature = util._compute_data_signature(image.data)
+
+            # Check for existing embeddings
+            if os.listdir(self.embeddings_save_path):
+                try:
+                    with zarr.open(self.embeddings_save_path, "a") as f:
+                        # If data_signature exists, compare and return validation message
+                        if "data_signature" in f.attrs:
+                            if img_signature != f.attrs["data_signature"]:
+                                return {
+                                    "message_type": "error",
+                                    "message": f"The embeddings don't match with the image: {img_signature} {f.attrs['data_signature']}"
+                                }
+
+                        # Load existing parameters
+                        if "tile_shape" in f.attrs:
+                            self.tile_x, self.tile_y = f.attrs["tile_shape"]
+                            self.halo_x, self.halo_y = f.attrs["halo"]
+                            self.model_type = f.attrs["model_type"]
```

Comment:
We also need to update these values in the GUI here. See https://github.com/computational-cell-analytics/micro-sam/blob/dev/micro_sam/sam_annotator/util.py#L671

Reply:
Nvm, you do this below.